File: tests/fixtures/swift/UserService.swift
import Foundation
import Combine

protocol UserRepository {
    func find(id: String) -> User?
    func save(user: User)
}

struct User {
    let id: String
    let name: String
}

enum Role {
    case admin
    case member
}

class UserService {
    let repo: UserRepository

    init(repo: UserRepository) {
        self.repo = repo
    }

    func find(id: String) -> User? {
        return repo.find(id: id)
    }

    func create(name: String) -> User {
        let user = User(id: UUID().uuidString, name: name)
        repo.save(user: user)
        return user
    }
}

func plainHelper(_ a: Int, _ b: Int) -> Int {
    return a + b
}

File: tests/fixtures/rust/user_service.rs
use std::collections::HashMap;

pub struct User {
    pub id: String,
    pub name: String,
}

pub trait Repository {
    fn find(&self, id: &str) -> Option<User>;
    fn save(&mut self, user: User);
}

pub struct InMemoryRepo {
    storage: HashMap<String, User>,
}

impl InMemoryRepo {
    pub fn new() -> Self {
        InMemoryRepo { storage: HashMap::new() }
    }
}

impl Repository for InMemoryRepo {
    fn find(&self, id: &str) -> Option<User> {
        self.storage.get(id).cloned().map(|u| User { id: u.id, name: u.name })
    }
    fn save(&mut self, user: User) {
        self.storage.insert(user.id.clone(), user);
    }
}

pub enum Role {
    Admin,
    Member,
}

pub fn plain_helper(a: i32, b: i32) -> i32 {
    a + b
}
